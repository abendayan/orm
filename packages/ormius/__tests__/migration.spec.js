const { Migration } = require('../lib/migration')

describe('migration', () => {
    test('setMigrationId', () => {
        const migration = new Migration()

        migration.setMigrationId('migrationTestId')
        expect(migration.migrationId).toBe('migrationTestId')
    })

    test('markAsRan', () => {
        const migration = new Migration()
        const connectionMock = { query: jest.fn() }

        migration.setMigrationId('migrationId')
        migration.setConnection(connectionMock)
        migration.markAsRan()
        expect(connectionMock.query.mock.calls).toMatchSnapshot()
    })

    test('markAsRan with error', () => {
        const migration = new Migration()
        const connectionMock = { query: jest.fn((query, migrationId, cb) => {
            cb('error')
        }) }

        migration.setMigrationId('migrationId')
        migration.setConnection(connectionMock)
        try {
            migration.markAsRan()
            expect(false).toBe(true)
        } catch (e) {
            expect(e).toBe('error')
        }
    })

    test('addColumn', () => {
        const migration = new Migration()

        migration.modelName = 'modelName'
        const connectionMock = { query: jest.fn() }

        migration.setMigrationId('migrationId')
        migration.setConnection(connectionMock)
        migration.addColumn('columnTest', 'string')
        expect(connectionMock.query.mock.calls).toMatchSnapshot()
    })

    test('addColumn with error', () => {
        const migration = new Migration()
        const connectionMock = { query: jest.fn((query, cb) => {
            cb('error')
        }) }

        migration.modelName = 'modelName'

        migration.setMigrationId('migrationId')
        migration.setConnection(connectionMock)
        try {
            migration.addColumn('columnTest', 'string')
            expect(false).toBe(true)
        } catch (e) {
            expect(e).toBe('error')
        }
    })

    test('createTable', () => {
        const migration = new Migration()

        migration.modelName = 'modelName'
        const connectionMock = { query: jest.fn() }

        migration.setMigrationId('migrationId')
        migration.setConnection(connectionMock)
        migration.createTable([{ type: 'string', name: 'column' }, { type: 'boolean', name: 'second_column' }])
        expect(connectionMock.query.mock.calls).toMatchSnapshot()
    })

    test('createTable with error', () => {
        const migration = new Migration()
        const connectionMock = { query: jest.fn((query, cb) => {
            cb('error')
        }) }

        migration.modelName = 'modelName'

        migration.setMigrationId('migrationId')
        migration.setConnection(connectionMock)
        try {
            migration.createTable([{ type: 'string', name: 'column' }, { type: 'boolean', name: 'second_column' }])
            expect(false).toBe(true)
        } catch (e) {
            expect(e).toBe('error')
        }
    })
})
