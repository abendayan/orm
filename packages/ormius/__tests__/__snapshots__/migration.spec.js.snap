// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`migration addColumn 1`] = `
Array [
  Array [
    "ALTER TABLE modelName ADD columnTest string",
    [Function],
  ],
]
`;

exports[`migration createTable 1`] = `
Array [
  Array [
    "CREATE TABLE modelName (column string,second_column boolean)",
    [Function],
  ],
]
`;

exports[`migration markAsRan 1`] = `
Array [
  Array [
    "INSERT INTO migrations (migration) VALUES (?)",
    "migrationId",
    [Function],
  ],
]
`;
